mirascope/core/base/_create.py

```diff
@@ -93,7 +94,7 @@ async def inner_async(
                 *args: _P.args, **kwargs: _P.kwargs
             ) -> TCallResponse | _ParsedOutputT:
                 fn_args = get_fn_args(fn, args, kwargs)
-                dynamic_config = await get_dynamic_configuration(fn, args, kwargs)
+                dynamic_config = await get_dynamic_configuration_async(fn, args, kwargs)
```

Comment:
If I understand correctly, await can only be called in the async function, so I split the function into two

Reply:
but `get_dynamic_configuration` can return an awaitable (i.e. by not calling await on the function) and defer calling `await` to the caller (which would be here where we're inside an async function). Does this make sense?

Reply:
Since the return value is expected to be `_BaseDynamicConfigT`, it is awaiting inside `get_dynamic_configuration_async` and returns list[BaseMessageParam] wrapped in dict like this.
If you don't await, the function returns `_BaseDynamicConfigT | list[BaseMessageParam` type.
```python
) -> _BaseDynamicConfigT:
...
    if _is_message_param_list(result):
        return cast(_BaseDynamicConfigT, {"messages": result})
```

Reply:
see my other comment on this